### PR TITLE
new module azure_rm_applicationsecuritygroup

### PR DESCRIPTION
**Pull request by yungezz**
new module azure_rm_applicationsecuritygroup. will update networkinterface module with application security group supported in seperate pr.

### CODE REVIEW

File: library/azure_rm_applicationsecuritygroup.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        foo: bar
+'''
+
+RETURN = '''
+id:
+    description:
+        - Resource id of the application security group.
+    returned: always
+    type: str
+    sample: /subscriptions/<subid>/resourceGroups/rg1/providers/Microsoft.Network/applicationSecurityGroups/MyAsg
```
Comment (zikalino) [inline library/azure_rm_applicationsecuritygroup.py]: just for consistency with passed parameters in examples, you could replace:
rg1 -> myRG

File: library/azure_rm_applicationsecuritygroup.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            if self.check_mode:
+                return self.results
+
+            response = self.create_update_applicationsecuritygroup()
+            self.results['id'] = response['id']
+
+        elif self.to_do == Actions.Delete:
+            self.log("Delete Application Security Group instance")
+            self.results['changed'] = True
+            self.results['id'] = old_response.get('id', None)
```
Comment (zikalino) [inline library/azure_rm_applicationsecuritygroup.py]: I am not sure about this.
I mean if instance is deleted, we should not return "id" any longer as it's no longer valid.
I think we have never considered that though....

Comment (yungezz) [reply library/azure_rm_applicationsecuritygroup.py]: this is good point. let me remove this.

File: library/azure_rm_applicationsecuritygroup.py (hunk 1 above)
Comment (yungezz) [reply library/azure_rm_applicationsecuritygroup.py]: fixed

**Pull request merged**